include/forkserver.h

```diff
@@ -146,6 +146,8 @@ typedef struct afl_forkserver {
   bool use_fauxsrv;                     /* Fauxsrv for non-forking targets? */
 
   bool qemu_mode;                       /* if running in qemu mode or not   */
+  
+  bool unicorn_mode;                    /* if running in unicorn mode or not*/
```

Comment:
Can we get around this extra flag? Ideally merge it with existing flags of the same meaning

Reply:
other modes have their own flag too - so why not?

Reply:
https://github.com/AFLplusplus/AFLplusplus/pull/2468/files#diff-d5539013e7d028b9fc3b6e581f284c5ebedc4cd8a7151b2c11cd0aeaabf071a0R42

We don't have a pointer to `afl` here so we must keep it to fsrv.

Reply:
> other modes have their own flag too - so why not?

The other modes should also not have extra flags, it should be a flag for feature xyz IMHO

Reply:
> > other modes have their own flag too - so why not?
> 
> The other modes should also not have extra flags, it should be a flag for feature xyz IMHO

then it is on us to refactor that for all of them, not in this PR - IMHO

Reply:
Fair